### PR TITLE
Document //flutter/runtime/dart_vm

### PR DESCRIPTION
**Pull request by chinmaygarde**
Related https://github.com/flutter/flutter/issues/42778

### CODE REVIEW

File: runtime/dart_vm.h [hunk 1]
```diff
@@ -26,22 +26,124 @@
 
 namespace flutter {
 
+//------------------------------------------------------------------------------
+/// @brief      Describes a running instance of the Dart VM. There may only be
+///             one running instance of the Dart VM in the process at any given
+///             time. The Dart VM may be created and destroyed on any thread.
+///             Typically, the first Flutter shell instance running in the
+///             process bootstraps the Dart VM in the process as it starts up.
+///             This cost is borne on he platform task runner of that first
```
Comment (jason-simmons) [inline runtime/dart_vm.h]: typo: "the"

File: runtime/dart_vm.h [hunk 2]
```diff
@@ -26,22 +26,124 @@
...
+///             available in the process, a new instance is created and a strong
+///             reference returned to the callers. The DartVMRef::Create call
+///             ensures that there are no data races during the creation or
+///             shutdown of a Dart VM (since a VM may be created and destroyed
+///             on any thread). Due to this behavior, all callers needing a
+///             running VM instance must provide snapshots and VM settings
+///             necessary to create a VM (even if they end up not being used).
+///
+///             In a running VM instance, the service isolate is launched by
+///             default is the VM is configured to do so. All root isolates must
```
Comment (jason-simmons) [inline runtime/dart_vm.h]: typo: "if"

File: runtime/dart_vm.h [hunk 3]
```diff
@@ -26,22 +26,124 @@
...
+  //----------------------------------------------------------------------------
+  /// @brief      The VM and isolate snapshots used by this running Dart VM
+  ///             instance.
+  ///
+  /// @return     A valid VM data instance.
+  ///
   std::shared_ptr<const DartVMData> GetVMData() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      The service protocol instance associated with this running VM
```
Comment (jason-simmons) [inline runtime/dart_vm.h]: typo: remove "VM"

File: runtime/dart_vm.h [hunk 4]
```diff
@@ -26,22 +26,124 @@
...
+  ///             isolate name server maps names (strings) to Dart ports.
+  ///             Running isolates can discover and communicate with each other
+  ///             by advertising and resolving ports at well known names.
+  ///
+  /// @return     The isolate name server.
+  ///
   std::shared_ptr<IsolateNameServer> GetIsolateNameServer() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      The task runner whose tasks may may be executed concurrently
```
Comment (jason-simmons) [inline runtime/dart_vm.h]: typo: remove second "may"

File: runtime/dart_vm.h [hunk 5]
```diff
@@ -26,22 +26,124 @@
...
+  ///             on a pool of worker threads. All subsystems within a running
+  ///             shell instance use this worker pool for their concurrent
+  ///             tasks. This also means that the concurrent worker pool may
+  ///             service tasks from multiple shell instances. The number of
+  ///             workers in a concurrent worker pool depending on the hardware
```
Comment (jason-simmons) [inline runtime/dart_vm.h]: typo: change "depending" to "depends"

File: runtime/dart_vm.h [hunk 6]
```diff
@@ -26,22 +26,124 @@
...
+  ///             concurrency of the target device (usually equal to the number
+  ///             of logical CPU cores).
+  ///
+  ///
+  /// @attention  Even though concurrent task queue is associated with a running
+  ///             Dart VM instance, the worker pool used to by the Flutter
```
Comment (jason-simmons) [inline runtime/dart_vm.h]: typo: remove "to"

File: runtime/dart_vm.h (hunk 1 above)
Comment (chinmaygarde) [reply runtime/dart_vm.h]: Done.

File: runtime/dart_vm.h (hunk 2 above)
Comment (chinmaygarde) [reply runtime/dart_vm.h]: Done.

File: runtime/dart_vm.h (hunk 3 above)
Comment (chinmaygarde) [reply runtime/dart_vm.h]: Done.

File: runtime/dart_vm.h (hunk 4 above)
Comment (chinmaygarde) [reply runtime/dart_vm.h]: Done.

File: runtime/dart_vm.h (hunk 5 above)
Comment (chinmaygarde) [reply runtime/dart_vm.h]: Done.

File: runtime/dart_vm.h (hunk 6 above)
Comment (chinmaygarde) [reply runtime/dart_vm.h]: Done.

**Comment by chinmaygarde** [comment]
The LUCI error has been fixed on ToT.

**Pull request merged**